torch/csrc/jit/passes/freeze_module.cpp

```diff
@@ -332,12 +395,32 @@ class AttributePropagator {
               auto attr = module.attr(name);
               insertMutableAttr(name, attr, mptr);
               if (attr.isModule()) {
-                modules.insert(attr.toModule()._ivalue());
+                auto module = attr.toModule();
+                if (modules.count(module._ivalue()) == 0) {
+                  if (moduleEscapes(module, graph)) {
+                    // Preserve all attributes and methods of 'module' type.
+                    auto type = module.type();
+                    attrsToKeep_[type].insert(type->numAttributes());
+                  }
+                  modules.insert(module._ivalue());
+                }
               }
               break;
             }
           }
+        } else if (n->kind() == prim::fork) {
+          applyToSubGraph(n, graph, [this](std::shared_ptr<Graph>& subgraph) {
+            recordReferencedAttrs(subgraph);
```

Comment:
Why not just directly call `recordReferencedAttrs` here ?

Reply:
Fork is not called with  self as param. 
_fork(submodule, foo)